crates/sdk/Cargo.toml

```diff
@@ -14,6 +14,12 @@ serde_json.workspace = true
 
 calimero-sdk-macros = { path = "./macros" }
 
+[target.'cfg(target_arch = "wasm32")'.dependencies]
+uuid = { version = "1.10.0", features = ["serde", "v5"] }
+
+[target.'cfg(not(target_arch = "wasm32"))'.dependencies]
+uuid = { version = "1.10.0", features = ["serde", "v4"] }
+
```

Comment:
^

Reply:
I will need to know a little more about your intentions here 🙂

Reply:
same as the snippet in `runtime`, this directive can be merged into `[dependencies]`, although following https://github.com/calimero-network/core/pull/799#discussion_r1800168580, it won't be necessary to have the dependency in the SDK at all

Reply:
Ahhh, I see. No - they are not the same thing. The WASM code cannot have the v4. In the other snippet, they were indeed the same, and I had overlooked that they had become so. Which means, that was a valid observation. But here, they are not the same.

Reply:
it doesn't need the `v5` either, it just needs to call `generate_uuid` which returns a UUID, so you don't need this directive

just make it

```toml
[dependencies]
uuid.workspace = true
```

but IMO (https://github.com/calimero-network/core/pull/799#discussion_r1800168580), this dependency doesn't even need to exist in the SDK at all

Reply:
It's not possible to do `uuid.workspace = true` here, as it doesn't exist in the workspace dependencies - it can't, due to this exact issue of arch targeting.

No problem removing the v5, although that was consciously there for use in tests, but that's not strictly necessary and not currently used.

Reply:
> It's not possible to do uuid.workspace = true here, as it doesn't exist in the workspace dependencies

it can be introduced

> it can't, due to this exact issue of arch targeting.

not true, with the version specified in the root manifest, you could do this in the local crate's

```rust
[target.'cfg(target_arch = "wasm32")'.dependencies]
uuid = { workspace = true, features = ["serde", "v5"] }

[target.'cfg(not(target_arch = "wasm32"))'.dependencies]
uuid = { workspace = true, features = ["serde", "v4"] }
```

Reply:
Okay, so that's fair about introducing it into the main workspace - my preference was not to do that, as I like it being obvious to have the arch matches together, but it's also fine having it in the main list and then overridden 👍